distributed/tests/test_worker_state_machine.py

```diff
@@ -40,6 +41,15 @@ async def wait_for_state(key: str, state: TaskStateState, dask_worker: Worker) -
         await asyncio.sleep(0.005)
 
 
+def test_task_state_tracking():
+    x = TaskState("x")
+    assert len(TaskState._instances) == 1
+    assert first(TaskState._instances) == x
+
+    del x
+    assert len(TaskState._instances) == 0
```

Comment:
You should add the `clean` ctxmanager or just the `clean_instances` ctxmanager. I don't think this is applied automatically and this test would otherwise be fragile and dependent on test order